tests/Behat/Context/Setup/ProductBundleContext.php

```diff
@@ -0,0 +1,183 @@
+use BitBag\SyliusProductBundlePlugin\Factory\ProductFactory;
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Formatter\StringInflector;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductTaxonInterface;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Sylius\Component\Product\Generator\SlugGeneratorInterface;
+use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class ProductBundleContext implements Context
+{
+    /** @var SharedStorageInterface */
+    private $sharedStorage;
+
+    /** @var FactoryInterface */
+    private $taxonFactory;
+
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
+
+    /** @var FactoryInterface */
+    private $productTaxonFactory;
+
+    /** @var EntityManagerInterface */
+    private $productTaxonManager;
+
+    /** @var ProductFactory */
+    private $productFactory;
+
+    /** @var FactoryInterface */
+    private $productBundleItemFactory;
+
+    /** @var FactoryInterface */
+    private $channelPricingFactory;
+
+    /** @var ProductVariantResolverInterface */
+    private $productVariantResolver;
+
+    /** @var SlugGeneratorInterface */
+    private $slugGenerator;
+
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        FactoryInterface $taxonFactory,
+        ProductRepositoryInterface $productRepository,
+        FactoryInterface $productTaxonFactory,
+        EntityManagerInterface $productTaxonManager,
+        ProductFactory $productFactory,
+        FactoryInterface $productBundleItemFactory,
+        FactoryInterface $channelPricingFactory,
+        ProductVariantResolverInterface $productVariantResolver,
+        SlugGeneratorInterface $slugGenerator
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->taxonFactory = $taxonFactory;
+        $this->productRepository = $productRepository;
+        $this->productTaxonFactory = $productTaxonFactory;
+        $this->productTaxonManager = $productTaxonManager;
+        $this->productFactory = $productFactory;
+        $this->productBundleItemFactory = $productBundleItemFactory;
+        $this->channelPricingFactory = $channelPricingFactory;
+        $this->productVariantResolver = $productVariantResolver;
+        $this->slugGenerator = $slugGenerator;
+    }
+
+    /**
+     * @Given /^the store has bundled product "([^"]*)" priced at ("[^"]+") which contains "([^"]*)" and "([^"]*)"$/
+     */
+    public function theStoreHasBundledProductPricedAtWhichContainsAnd(string $productBundleName, int $productBundlePrice, string $firstProductName, string $secondProductName)
```

Comment:
Break line? Didn't our tools fix it?

Reply:
I don't know why, I've fixed it manually.